### PR TITLE
Implement web3 react

### PR DESCRIPTION
**Pull request by RamRamez**
1. implement web3 react for connecting wallets
2. implement select wallet modal
3. implement user provider context
4. add hearted state to project cards and single project view
5. implement switch wallet
6. implement wallet menu details
7. implement fetching balance
8. implement primary and secondary networks
9. implement fetching token signer function

### CODE REVIEW

File: config.ts [hunk 1]
```diff
@@ -2,11 +2,27 @@ const config = {
   QRAPHQL_SERVER: 'https://serve.giveth.io/graphql',
   APP_URL: 'https://typescript.giveth.io/',
   REPORT_ISSUE_URL: 'https://github.com/Giveth/giveth-io-typescript/issues/new',
-  GIVeconomy_URL: 'https://liquidity-mining-dapp.vercel.app/'
+  GIVeconomy_URL: 'https://liquidity-mining-dapp.vercel.app/',
+  OUR_SECRET: 'our_secret_message_string',
```
Comment (mateodaza) [inline config.ts]: I know we are not using OUR_SECRET at the moment but as soon as we start doing mutations we will. Let's move this to the env file

File: pages/partnerships.tsx [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import React from 'react'
 import Footer from '../src/components/Footer'
-import Menubar from '../src/components/Menubar'
+import MenuInedx from '../src/components/menu/MenuInedx'
```
Comment (mateodaza) [inline pages/partnerships.tsx]: Misspelled Index

File: pages/project/[slug].tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { FETCH_PROJECT_BY_SLUG } from '../../src/apollo/gql/gqlProjects'
 import client from '../../src/apollo/apolloClient'
 import { IProjectBySlug } from '../../src/types/types_graphql'
 import Head from 'next/head'
-import Menubar from '../../src/components/Menubar'
+import MenuInedx from '../../src/components/menu/MenuInedx'
```
Comment (mateodaza) [inline pages/project/[slug].tsx]: Misspelled Index here too

Comment (mateodaza) [reply pages/project/[slug].tsx]: And like three other files

**Review by mateodaza — CHANGES_REQUESTED** [review]

**Comment by mateodaza** [comment]
I remember you said we can start merging this little by little, should we merge this and then create another PR to the missing parts?

File: config.ts (hunk 1 above)
Comment (RamRamez) [reply config.ts]: I copied that from the env file

**Comment by RamRamez** [comment]
> I remember you said we can start merging this little by little, should we merge this and then create another PR to the missing parts?

Yes, true.

**Pull request merged**